src/JavaCodeGenerator.cs

```diff
@@ -3114,1874 +3587,3434 @@ private enum MethodType
+                            }
+                            return result;
+                        });
 
-                foreach (ParameterMapping mapping in transformationParameterMappings)
-                {
-                    string inputPath = GetParameterName(mapping.InputParameter);
-                    if (mapping.InputParameterProperty != null)
+                    if (nextMethodGroup == null || autoRestMethod.Group == nextMethod.Group)
                     {
-                        inputPath += "." + CodeNamer.Instance.CamelCase(mapping.InputParameterProperty) + "()";
+                        nextMethodInvocation = nextMethodName;
                     }
-                    if (filterRequired && !mapping.InputParameter.IsRequired)
+                    else
                     {
-                        inputPath = "null";
+                        nextMethodInvocation = $"{(autoRestMethod.Group.IsNullOrEmpty() ? "this" : "client")}.get{nextMethodGroup.ToPascalCase()}().{nextMethodName}";
                     }
+                }
+
+                string nextPageLinkParameterName = null;
+                string nextGroupTypeName = null;
+                string groupedTypeName = null;
+                string nextMethodParameterInvocation = null;
+                AutoRestParameter groupedType = null;
+                if (nextMethod != null)
+                {
+                    nextPageLinkParameterName = nextMethod.Parameters
+                        .Select((AutoRestParameter parameter) =>
+                        {
+                            string parameterName;
+                            if (!parameter.IsClientProperty)
+                            {
+                                parameterName = parameter.Name;
+                            }
+                            else
+                            {
+                                string caller = (parameter.Method != null && parameter.Method.Group.IsNullOrEmpty() ? "this" : "this.client");
+                                string clientPropertyName = parameter.ClientProperty?.Name?.ToString();
+                                if (!string.IsNullOrEmpty(clientPropertyName))
+                                {
+                                    CodeNamer codeNamer = CodeNamer.Instance;
+                                    clientPropertyName = codeNamer.CamelCase(codeNamer.RemoveInvalidCharacters(clientPropertyName));
+                                }
+                                parameterName = $"{caller}.{clientPropertyName}()";
+                            }
+                            return parameterName;
+                        })
+                        .First((string parameterName) => parameterName.StartsWith("next", StringComparison.OrdinalIgnoreCase));
+
+                    IEnumerable<AutoRestParameter> nextMethodRestAPIParameters = nextMethod.Parameters
+                        .Where((Au
```

Comment:
So does this `type` parameter pretty much determine which method body gets generated?

Reply:
Correct.

Reply:
It also is used to determine which imports the ClientMethod object will add to the imports set (see ClientMethod.java for more on that).